frontend/src/components/class/ClassList.tsx

```diff
@@ -1,43 +1,47 @@
-import {
-  DataTablePageEvent,
-  DataTableSortEvent,
-  DataTableFilterEvent,
-} from "primereact/datatable";
-import { Column } from "primereact/column";
-import { useCallback, useState } from "react";
+import { useState } from "react";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faEdit } from "@fortawesome/free-regular-svg-icons";
 import { defineMessages, FormattedMessage, useIntl } from "react-intl";
 import styled from "@emotion/styled";
 import { faAdd } from "@fortawesome/free-solid-svg-icons";
 import { useRouter } from "next/router";
-import DataTable, { LazyTableState } from "@/components/DataTable";
-import { getClassStatusMessage } from "@/i18n/class-status-messages";
-import { TableMessages } from "@/i18n/table-messages";
-import { ExistingClassWithTeacher } from "@/api/collimator/models/classes/existing-class-with-teacher";
-import { useDeleteClass } from "@/api/collimator/hooks/classes/useDeleteClass";
-import { useAllClassesLazyTable } from "@/api/collimator/hooks/classes/useAllClasses";
+import { ColumnDef } from "@tanstack/react-table";
 import ConfirmationModal from "../modals/ConfirmationModal";
 import SwrContent from "../SwrContent";
 import Button, { ButtonVariant } from "../Button";
 import DropdownMenu from "../DropdownMenu";
+import ChakraDataTable from "@/components/ChakraDataTable";
+import { getClassStatusMessage } from "@/i18n/class-status-messages";
+import { TableMessages } from "@/i18n/table-messages";
+import { ExistingClassWithTeacher } from "@/api/collimator/models/classes/existing-class-with-teacher";
+import { useDeleteClass } from "@/api/collimator/hooks/classes/useDeleteClass";
+import { useAllClasses } from "@/api/collimator/hooks/classes/useAllClasses";
+import { ColumnType } from "@/types/tanstack-types";
 
 const ClassListWrapper = styled.div`
   margin: 1rem 0;
-
-  tr {
-    cursor: pointer;
-  }
 `;
 
 const messages = defineMessages({
   nameColumn: {
     id: "ClassList.columns.name",
     defaultMessage: "Name",
   },
-  lastSessionColumn: {
-    id: "ClassList.columns.lastSession",
-    defaultMessage: "Last Session",
+  degreeColumn: {
+    id: "ClassList.columns.degree",
+    defaultMessage: "Degree",
+  },
+  createClass: {
+    id: "ClassList.createClass",
+    defaultMessage: "Create Class",
+  },
+  schoolYearColumn: {
+    id: "ClassList.columns.schoolYear",
+    defaultMessage: "School Year",
+  },
+  teacherColumn: {
+    id: "ClassList.columns.teacher",
+    defaultMessage: "Teacher",
```

Comment:
Per our discussion on Nov 6, we don't need these extra columns.
(Although we could keep the Teacher I guess)